compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/SkikoUIView.kt

```diff
@@ -301,34 +287,43 @@ internal class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol {
 
         _touchesCount += touches.size
 
-        sendTouchEventToSkikoView(withEvent!!, SkikoPointerEventKind.DOWN)
+        withEvent?.let {
+            delegate?.onPointerEvent(it.toSkikoPointerEvent(SkikoPointerEventKind.DOWN))
+        }
     }
 
     override fun touchesEnded(touches: Set<*>, withEvent: UIEvent?) {
         super.touchesEnded(touches, withEvent)
 
         _touchesCount -= touches.size
 
-        sendTouchEventToSkikoView(withEvent!!, SkikoPointerEventKind.UP)
+        withEvent?.let {
+            delegate?.onPointerEvent(it.toSkikoPointerEvent(SkikoPointerEventKind.UP))
+        }
     }
 
     override fun touchesMoved(touches: Set<*>, withEvent: UIEvent?) {
         super.touchesMoved(touches, withEvent)
-        sendTouchEventToSkikoView(withEvent!!, SkikoPointerEventKind.MOVE)
+
+        withEvent?.let {
+            delegate?.onPointerEvent(it.toSkikoPointerEvent(SkikoPointerEventKind.MOVE))
+        }
     }
 
     override fun touchesCancelled(touches: Set<*>, withEvent: UIEvent?) {
         super.touchesCancelled(touches, withEvent)
 
         _touchesCount -= touches.size
 
-        sendTouchEventToSkikoView(withEvent!!, SkikoPointerEventKind.UP)
+        withEvent?.let {
+            delegate?.onPointerEvent(it.toSkikoPointerEvent(SkikoPointerEventKind.UP))
+        }
     }
 
-    private fun sendTouchEventToSkikoView(event: UIEvent, kind: SkikoPointerEventKind) {
-        val pointers = event.touchesForView(this).orEmpty().map {
+    private fun UIEvent.toSkikoPointerEvent(kind: SkikoPointerEventKind): SkikoPointerEvent {
+        val pointers = touchesForView(this@SkikoUIView).orEmpty().map {
```

Comment:
Why do you choose extension function?
From my point of view it is better to leave simple function because of `this` link.
But it is not a my strong opinion, you may leave as is.

Reply:
I think it's good that way because it indicates SkikoPointerEvent <=> UIEvent relationship. Hints well at possible removal of SkikoPointerEvent here, because we have redundant conversion every single time:

UIEvent -> SkikoPointerEvent -> (Compose)PointerEvent

Reply:
Ok, fine!